Assets/Scripts/Door.cs

```diff
@@ -0,0 +1,21 @@
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Door : MonoBehaviour, IInteractable
+{
+    [SerializeField] private Object sceneToLoad;
+
+    public void Interact()
+    {
+        if (sceneToLoad != null)
+        {
+            // if we are in hunting scene, call respawn in house instead
+            if (FindObjectOfType<HuntingManager>() != null)
+                HuntingManager.Instance.RespawnInHouse();
+            else
```

Comment:
Are these lines actually needed? We could just drag the house scene to the field instead, preventing a need for the expensive `FindObjectOfType` call

Reply:
I use `FindObjectOfType` because I wanted to check whether the player is in the hunting scene or not, since switching from hunting scene to house scene will merge the hunting inventory with the permanent inventory (which is done in the `RespawnInHouse` function. The other option that I thought of was just to check the scene name, but it will break if we rename a scene without also changing the code.

Reply:
Since it's a singleton, could we check instead for `HuntingManager.Instance` is `null` instead?

Reply:
Yeah that should work. Dont know why I havent thought abt this. Fixed.